python/dgl/data/knowledge_graph.py

```diff
@@ -0,0 +1,582 @@
+from __future__ import absolute_import
+
+import numpy as np
+import pickle as pkl
+import networkx as nx
+import scipy.sparse as sp
+import os, sys
+
+from .dgl_dataset import DGLBuiltinDataset
+from .utils import download, extract_archive, get_download_dir
+from .utils import save_graphs, load_graphs, save_info, load_info, makedirs, _get_dgl_url
+from .utils import generate_mask_tensor
+from .utils import deprecate_property, deprecate_function
+from ..utils import retry_method_with_fix
+from .. import backend as F
+from ..graph import DGLGraph
+from ..graph import batch as graph_batch
+from ..convert import graph as dgl_graph
+
+class RGCNLinkDataset(DGLBuiltinDataset):
```

Comment:
Will there be more datasets in this format? If so, I'd like to discuss if we could single this class out in a separate file.